task_manager/src/lib.rs

```diff
@@ -167,17 +16,17 @@ use rusqlite::{
 };
 use serde::Serialize;
 
+// mod adv_sqlite;
+mod mem_db;
+
 // Types
 // ----------------------------------------------------------------
-
-#[derive(Debug, thiserror::Error)]
+#[derive(PartialEq, Debug, thiserror::Error)]
 pub enum TaskManagerError {
     #[error("IO Error {0}")]
     IOError(IOErrorKind),
     #[error("SQL Error {0}")]
     SqlError(String),
-    #[error("Serde Error {0}")]
-    SerdeError(#[from] serde_json::error::Error),
 }
 
 pub type TaskManagerResult<T> = Result<T, TaskManagerError>;
```

Comment:
This now seems duplicated with the [`adv_sqlite.rs`](https://github.com/taikoxyz/raiko/blob/219b4556588def1e4b50b7d46a84b6909202e4df/task_manager/src/adv_sqlite.rs) file

Reply:
Yes, the problem is I stuck in some lifetime compile issues when I start modify adv_sqlite (previous sqlite), so here for idea demo, I just keep the adv_sqlite as it was. If this change is good to you guys, I will refine it again.

Reply:
Seems good to me, although I believe that the changes to the trait functions I made on the sqlite driver are more usable given the params we receive in the request on the host. For example we need to do a RPC call to get the blockhash whereas we receive the block number in the request parameters. So changing that to needing just the block number removes the need to fetch before starting work on a proof. This, in turn, allows us to just mark the request as submitted, then when a worker picks up the task the worker can fire a request to get the additional details, thus saving us a RPC fetch on both task submission and task querying. This by extensions makes our API less prone to failures due to RPC issues and faster since we are only fetching data from memory or at worst from our own disk in the future.

Reply:
Yes, I see your point. I was thinking proofrequest misses lots of data which needs to be fetched from other places. My init plan was to make a conversion like `From<ProofRequest> for EnqueueParam` which is called before enqueue task, and ideally at that position, all required data are presented(either fetched or cached somewhere). However, when using proofrequest, either we add those absent info into fn input, like previous chain_id (no real problem actually, the potential diff exists in only if we want to add more, i.e., change structure seems little bit easier than change fn declaration), or we fetch it again inside task manager (not a good choice).

BTW: block-hash is a special case because we always need to fetch to avoid a incorrect header introduced by re-org (although highly unlikely). For all of the rest, we can carefully choose them to make sure there is no redundant RPCs.

Reply:
So maybe we make the following params required for a task enqueue:
- `chain_id`
- `blockhash`
- `proofsys_id`
- `prover`
- `request` (optional as it is just the json serialized proof request object)

Reply:
We could then optimize the RPCs by only fetching the block header in to get the blockhash instead of also fetching the blob before it is needed?

Reply:
Yes, exactly, I will make it pass CI ASAP, then, we can refine it to production level.